### DISCUSSION TITLE
Flat Keyboard

### DISCUSSION THREAD

**Discussion by Dakkaron** [Dakkaron/Fairberry#8] [Ideas]
As @dumblob requested, here is a mockup for a flat keyboard, that doesn't overlap the phone. It is raised by about 2mm over the phone, to leave a little bit more space for the USB C connector. Also, a bit of extra height is necessary for the bracket

I can imagine this working out, especially if you have a smaller and lighter phone. The phone in this render is the (very tall and heavy) Fairphone 4.

![Diagonal render](https://user-images.githubusercontent.com/6311558/167219866-4dfa1112-f205-4c25-ac83-8f36ee7e2b15.png)
![Top down wireframe](https://user-images.githubusercontent.com/6311558/167219807-4fa40cf1-d223-4ba5-85b4-5ce409061ee9.png)
![Side](https://user-images.githubusercontent.com/6311558/167219909-3fcc1b0d-8637-4ec7-8701-2e6975b4c029.png)

**Comment by dumblob** [comment]
LGTM - any sources or at least an STL to see some more details?

**Reply by Dakkaron** [reply]
I just uploaded the files here: https://github.com/Dakkaron/Fairberry/tree/main/Case/Fairphone4/Flat-Case

Yes, I know, I shouldn't be doing stuff like this in Blender... I will remake the case in some real CAD software.

**Reply by dumblob** [reply]
Blender is actually very good for such stuff. No need to remake anything IMHO (at least not now).

Thanks for the files, I'll look at them soon.

**Comment by ghost** [comment]
I think this looks fairly decent. It's my favorite design yet!
Is it possible to have this as some sort of case?
I think it would look more decent and will probably a bit more stable.

>I can imagine this working out, especially if you have a smaller and lighter phone. The phone in this render is the (very tall and heavy) Fairphone 4.

Tbh I don't care about the thickness. As an ex-KeyOne user. I've been using one of those rings on the back, which helped a lot and now I even use one for an old iPhone SE (2016) which is way smaller and it's just so good. So I believe it might just work out with any other phone.

**Reply by Dakkaron** [reply]
It's printing right now, I'll post pictures later.

The iPhone SE should have a good form factor for this. The only issue is the missing USB port. I guess, for this to work with a Lightning connector, you'd have to cut apart a Lightning to USB cable and see, if you can fit that into the case. If you are lucky, the conversion between the two protocols happens in the lightning connector part (meaning the signals in the cable are already USB. That way you can just chop off the USB connector and solder the cable to the board.

I guess, you might have the best chance with an USB C cable, since there you don't have a massive USB A connector, where they most likely put all the conversion circuits. USB C isn't much bigger than Lightning, so chances are better for the conversion to happen on the Lightning side.

**Reply by Dakkaron** [reply]
#
![20220507_134453](https://user-images.githubusercontent.com/6311558/167253051-ddda7066-6a6d-4273-bd3f-d15b51d68e7b.jpg)
![20220507_134525](https://user-images.githubusercontent.com/6311558/167253054-09d058fb-4d21-42e5-b796-77f212cdd06a.jpg)
![20220507_134502](https://user-images.githubusercontent.com/6311558/167253079-ba94bff1-77db-497b-a570-2708d0e5483f.jpg)
![20220507_134514](https://user-images.githubusercontent.com/6311558/167253080-d896eb49-5a8a-4675-bb8c-fda752b465a1.jpg)

Sorry for the bad print quality, I forgot, what a terrible material that wood filament is.

**Reply by Dakkaron** [reply]
Tbh, I like the overlapping design better. It feels better in the hand (or maybe I am just more used to that design). That said, the flat design isn't terrible either.

This is what I currently use:

![20220507_134650](https://user-images.githubusercontent.com/6311558/167253261-d9df20a2-f463-406b-ae5b-d714bbd71adc.jpg)
![20220507_134637](https://user-images.githubusercontent.com/6311558/167253265-0ed2d48d-3942-4e24-948f-3d276fabaaa0.jpg)
![20220507_134704](https://user-images.githubusercontent.com/6311558/167253271-06a626cc-3675-4d92-9cf8-8ee0deafe802.jpg)

**Reply by ghost** [reply]
It doesn't look half bad for a prototype tbh. It's quite tall in total, but I like it. Good job!

**Reply by dumblob** [reply]
I really like how this tall design turned out. I definitely will give it a shot once I (or someone quicker than me) will make a whole case from it.

It actually seems you've glued together some DIY "top-half case" with your overlapping design. Is that so?

From my limited CAD experience this "flat keyboard" shall allow make a whole case with a keyboard as one 3D printed piece. Do you also see it this way?

**Reply by Dakkaron** [reply]
For the red/white design I actually chopped up a 3D printed case made from TPE that I had lying around, and joined the pieces using an old soldering iron. TPE is a very rubber-like flexible material, that allows me to easily remove the case. I made this one mostly to see if the two-material thing does actually work in practice, and it does.

I upgraded my 3D printer recently to a parking dual extruder design, so that I can print with two materials at the same time. So I'll try to make a case like that in a single print.

For regular single-extruder 3D printers I am planning to make a design where both parts are joined using something like a dovetail joint, which can then be welded together using an old soldering iron. Since TPE sticks incredibly well to PLA or PETG, I believe it shouldn't be a problem at all.

I don't see a way though, to print a full case in one piece. At least not in rigid material like PLA. You still have to get the case over the edge of the phone, preferably without damaging the volume buttons.

Also, a last observation: Since the device currently doesn't do pass-through charging, there needs to be either a way to easily detach the keyboard or to at least plug the cable into the USB port without taking off the keyboard. Because of that, I wouldn't do a permanent attachment at this point.

I also have ideas for a case where the top half stays on the device, but the bottom half clicks into the top half, so that the keyboard can easily be removed.

If you have an idea how to get a good single-piece case, can you maybe draw something up? (Doesn't have to be CAD, Paint is good enough for concepts)

**Reply by dumblob** [reply]
![smartphone-case-flexible00](https://user-images.githubusercontent.com/2451044/167852918-64628c1a-6a3c-41ee-b994-599c1cbdb2e0.jpg)

![smartphone-case-flexible00-view2](https://user-images.githubusercontent.com/2451044/167852994-529de503-3947-4e6f-af19-bcb962d7b191.jpg)

Something like this could make it flexible enough.

We can also get more creative:

![smartphone-case-flexible01-view0](https://user-images.githubusercontent.com/2451044/167856669-20b3a026-6f25-45a8-a10c-549bfc37b0af.jpg)

![smartphone-case-flexible01-view1](https://user-images.githubusercontent.com/2451044/167856711-c955f260-4d6e-408a-b9e5-5c9b3a75c477.jpg)

or even more creative... :stuck_out_tongue:

**Reply by dumblob** [reply]
Or just cut a really narrow cutout into the top like this:

![smartphone-case-flexible02-view0](https://user-images.githubusercontent.com/2451044/167860049-319a9891-508f-4983-a3f6-0a12ba0b9754.jpg)

This shall maintain acceptable sturdiness even in corners without affecting the overall look much.

Note this is bad engineering - the hole shall have looked like this actually (rule no. 1 is "no sharp edges"):

![smartphone-case-flexible02-view1](https://user-images.githubusercontent.com/2451044/167863540-f3a10aae-2257-40e9-ba70-37f058ea7f0b.jpg)

![smartphone-case-flexible02-view2](https://user-images.githubusercontent.com/2451044/167863581-71d89ed3-1c0e-46c8-acd1-692d4b4db439.jpg)

**Reply by Dakkaron** [reply]
Really cool ideas! Thanks for CADing that! Regarding the sharp corners: On a 3D print this doesn't really matter, since you never actually have sharp corners.

But you are right, this might make a single-piece print viable.

**Comment by Dakkaron** [comment]
I made a quick mockup in black with a quick and dirty demo bumper case for the top, just to show what it might look like when finished.

![20220508_203109](https://user-images.githubusercontent.com/6311558/167439276-5ae6022d-9b9b-412b-a7d1-3b648862e813.jpg)
![20220508_203047](https://user-images.githubusercontent.com/6311558/167439281-991b554e-e739-496b-92d4-ab93fd7b84fd.jpg)

Here is a size comparison to the original version with the keyboard stacked on top of the phone.
![20220508_204029](https://user-images.githubusercontent.com/6311558/167439262-d881e0e5-d5a0-4551-ae91-e913b4844410.jpg)

**Reply by dumblob** [reply]
Oh man you're quick. I don't know why, but I absolutely love what I see!

I have one more person who would welcome having such keyboard. All this effort will pay out!

Yep, I'll try to quickly draw what I meant. The idea is (yes, again :smile:) to make certain parts of the case flexible but in a way that they're still sturdy enough to absorb hits from (small) falls.

**Comment by spARTacusFromCB** [comment]
Not having the keyboard overlap the phone screen is what I assumed I would desire, in order to get the surface plane of the keyboard almost even with the surface of the screen. I guess the overall arrangement does get long, especially on a big phone. However, I assume if someone wants a PKB attachment on their phone, they aren't fussing over the fact that they are making their phone longer. That said, if the surface of the keyboard MUST be higher than the surface of the phone to accommodate the plane of the USB and also all the electronics, I guess most would also consider sacrificing some screen space when the keyboard is attached, since as it stands now a big chunk of the phone screen is consumed anyway with the VKB. So many things now that I didn't think I was going to have to reconsider.

**Comment by spARTacusFromCB** [comment]
I wonder how far over the screen one could put the keyboard (shrinking the screen via the developer options command). Theoretically, could one completely overlap the entire keyboard above the phone given the length of the flex ribbon? My current phone is quite big compared to my last phone and since the VKB normally always consumes lots of screen, maybe the entire keyboard over the screen is something to think about.

**Comment by spARTacusFromCB** [comment]
Hey I have a new idea. How a about a slip around attachment? Basically, a hollow contained arrangement that slips around the bottom portion of the phone, with the keyboard ending up over the the phone screen and the breakout plus arduino boards ending up behind the phone, then also a 90 degree angle USB connector in the middle.

**Reply by dumblob** [reply]
I have actually thought of this some months ago and this is what I immediately found out.

It's super complicated, thus:

  1. (this is subjective) both of the two constructions variants I can think of would actually look cool - maybe this could be made into a steam-punk smartphone!
  2. more parts to engineer & manufacture
  3. quite bulky - in an "opened" state you'll end up having one "layer" (3-5 mm) on top of the display, then another at the opposite side on the back (3-5 mm) and one at the bottom of the phone (3-5 mm)
  4. more difficult to port to different phones
  5. more difficult to put on and put off - thus requiring a pass-through USB (i.e. also data lanes - not just vcc for charging!)
  6. fragile - moving parts are the worst evil of all evil (still remember how [giants got scared by foldables](https://www.youtube.com/watch?v=MvadqG6R_7M&t=0s )?)
  7. one of the two construction variants I can think of will even slightly more destabilize the phone in your hands due to more mass in the upper half
  8. the other variant would though form an inevitable "stair" which would make it difficult to put into your pocket
  9. more expensive than (any) designs without moving parts

But this might be off as I'm describing what I had in mind and not what you had in mind :wink:. So take it with a grain of salt.

Would you like to maybe post some visualizations of what you imagine (mspaint is good enough)?

**Reply by spARTacusFromCB** [reply]
The bottom might not have to be enclosed. If using a 90 degree connector to plug the USB into the phone. Something as simple as band (elastic/stretchy) around the phone with the keyboard on the top and the boards on the back could potentially even suffice to "hold" the keyboard in place. It would probably only work if the entire keyboard could overlap over the phone (which might not be possible either). I once used a broccoli band for a wallet and my credit cards. Something like this ( https://grifiti.com/products/grifiti-band-joes-3-25-x-1-25-in-standard-bands-5pk  ), which made me wonder about it for holding the keyboard on the front and the boards on the back. It was just a thought. Most certainly not as slick as a 3D printed case.

Here is

**Reply by Dakkaron** [reply]
If I visualize that correctly, that would mean, you'd have to put the band over the keyboard, which would obstruct the keys...

Also putting the boards on the back means, you need to wrap the fragile flex PCB cable of the keyboard to the rear. That thing is really delicate. I wouldn't stress it more than necessary. I got a new 3D printed case though, that I just finished yesterday, and it's really good. I'd go as far as to call it version 1.0.

The USB connector is now screwed to the case, so that it's plugged/unplugged by sliding the attachment on/off. This removes the possible durability issues of the solder joints, since there are actually no moving parts anymore. Also, it removes the need for pass-through charging, since it's really easy to remove the keyboard to free up the USB port.

Also, I made a fitting hard top case, which is locked in place by the camera bump.

I will post aome pictures and maybe a video soon.

**Reply by spARTacusFromCB** [reply]
Yeah the band idea was really just a crazy thought. If it could be done, I think the keyboard (back of it) would need to be attached to the band. I figured the flex ribbon would be a limitation (not being able to get around to the back of the phone). It was just a crazy thought. Can't wait to see your latest case version.

**Reply by dumblob** [reply]
Also can not wait to see the new split version (top & bottom halves separate). Also if the bottom half is really precisely printed and all dimensions are totally correct, then the screw solution for the USB connector is definitely the way to go. Or maybe use a flexible screw from rubber/TPE to allow for vibrations absorption?

Btw. does the camera bump (incl. camera protection screen) not get weared off quickly due to (potentially frequent) (de)attaching of the keyboard?

I am still interested in a case from "one piece" - so I will see how this all evolves :wink:.

Really excited about this all!

**Reply by Dakkaron** [reply]
![20220516_103313](https://user-images.githubusercontent.com/6311558/168626952-0c4d4598-99e2-4601-9dd8-d49239ee0ed0.jpg)
![20220516_103321](https://user-images.githubusercontent.com/6311558/168626960-c2cd42eb-c1a4-414d-a86f-94cc2404fc3a.jpg)
![20220516_103333](https://user-images.githubusercontent.com/6311558/168626971-ce5f8160-65a5-413f-bee3-1d69ad5f332f.jpg)

When detaching the keyboard, the top part stays on the phone, still providing protection.

The only place where I expect some wear is the bottom part, where the keyboard sits, since I don't frequently remove the top part. It's actually kind of hard to remove.

The USB plug position actually has quite a bit of tolerance. USB C is made to be plugged in with fingers, so the tolerances of a decent 3D printer are far tighter than that. I printed that thing first on my really inaccurate, old 3D printer and it works flawlessly. I didn't have a single time where it didn't work. Granted, I only used it for a few days, but I've plugged/unplugged it quite a few times now. Also, unless you ram it on with force, there isn't anything that could really be damaged here, since it's all quite strong. Worst case, the 3D printed part fails, and that's not much of an issue.

**Reply by spARTacusFromCB** [reply]
Looks really good. There seems to be a fair bit of room still also left inside the "lower case", behind the male USB connector and beneath the keyboard (I can't even see breakout and Arduino boards). Presumably, this concept (the USB connector being fixed in place to the "lower case" would also work for if someone wanted the keyboard surface plane even lower or for not overlapping the keyboard over the screen?

**Reply by spARTacusFromCB** [reply]
Another benefit of this approach is people could probably just still buy their own normal full phone case from Amazon or where ever they prefer, and cut it off horizontally to use as the "top case" part, in combination with a 3D printed one for the lower case part containing the keyboard. Then, they'd still also have a left over a "lower case" part to use whenever they remove the keyboard.

**Reply by Dakkaron** [reply]
> Looks really good. There seems to be a fair bit of room still also left inside the "lower case", behind the male USB connector and beneath the keyboard (I can't even see breakout and Arduino boards). Presumably, this concept (the USB connector being fixed in place to the "lower case" would also work for if someone wanted the keyboard surface plane even lower or for not overlapping the keyboard over the screen?

I've been looking into it today. If we use a different Arduino (it's called Beetle: https://www.amazon.de/dp/B07JK5RKQ4) it would be possible to fit everything in that bottom space. But for that to work, the keyboard flex PCB would have to make an uncomfortable bend... I am not too happy with that arrangement.

On the other hand, that way the keyboard could sit almost flash against the screen (only separated by some layers of tape, so the metal of the backside of the keyboard doesn't touch the screen, which would trigger ghost touches).

That would slim the whole thing down to 16.5mm total height, which is pretty slim.

The Beetle board might be the better choice anyways, since it's not necessary to remove a USB port and also the pads where the USB plug would have to be soldered to are much bigger. So it's a lot easier to solder that one.

The only disadvantage it has is fewer pins. So there are no pins left for further expansion.

> Another benefit of this approach is people could probably just still buy their own normal full phone case from Amazon or where ever they prefer, and cut it off horizontally to use as the "top case" part, in combination with a 3D printed one for the lower case part containing the keyboard. Then, they'd still also have a left over a "lower case" part to use whenever they remove the keyboard.

True. There is actually no real benefit to a printed to half (except cost).

I also thought of making a dummy bottom part, but since my keyboard is on almost all of the time, I didn't bother yet.

**Reply by spARTacusFromCB** [reply]
Wow that beettle looks small. I guess it is an Arduino clone, like also the Micro Pro? It says it isn't available (maybe just not for shopping to CA).

**Reply by Dakkaron** [reply]
Yeah, it's on Amazon DE and it's available for me, but I guess, that's because I am in Europe. Watch out when buying beetles, there are many different versions, since, as you said, it's not an official board. There is one version that's even smaller but has too few pins, there is a version that uses the ATTiny85 (which won't work, too few pins and no hardware USB).

And then there are two versions of the correct thing: The 3.3V version and the 5V version. I'd get the 5V version, but both work. With the 5V version, power consumption might be a bit higher (but maybe not, because it leaves off one step of DCDC power conversion), but it will allow for a brighter maximum keyboard brightness.

5V version: https://www.amazon.ca/dp/B07KF9QXPX/
3.3V version: https://www.amazon.ca/dp/B09LVFRRRN/

**Reply by dumblob** [reply]
> I've been looking into it today. If we use a different Arduino (it's called Beetle: https://www.amazon.de/dp/B07JK5RKQ4) it would be possible to fit everything in that bottom space. But for that to work, the keyboard flex PCB would have to make an uncomfortable bend... I am not too happy with that arrangement.

Hm, could you make 3D or 2D drawing of the problem? I bet there is some way to make the radius of the bend larger without affecting the slimness of the keyboard :wink:. I will take a look and propose some ideas.

> That would slim the whole thing down to 16.5mm total height, which is pretty slim.

Sounds good to me.

> The Beetle board might be the better choice anyways, since it's not necessary to remove a USB port and also the pads where the USB plug would have to be soldered to are much bigger. So it's a lot easier to solder that one.

:+1:

> I also thought of making a dummy bottom part, but since my keyboard is on almost all of the time, I didn't bother yet.

I for myself wouldn't have any use for a dummy bottom part :wink:.

And thanks for checking all the details about arduinos - that's very important to know!

**Reply by Dakkaron** [reply]
> Hm, could you make 3D or 2D drawing of the problem? I bet there is some way to make the radius of the bend larger without affecting the slimness of the keyboard 😉. I will take a look and propose some ideas.

The biggest problem with the bend would be, that it's not a straight 2D-bend anymore. It would twist the flex PCB in all 3 dimensions. It's a bit hard to describe.

I today finished the design of a new PCB that should solve "all" the problems. I'll wait a few days, look over the design again to check for errors, and then I'll order some to see if my concept works. This also removes all those free-flying wires that the current solution uses, so it should increase durability and reliability.

> And thanks for checking all the details about arduinos - that's very important to know!

No worries! Sadly, there are no smaller Arduinos with an ATMega32u4 and enough pins...

**Comment by theblcl** [comment]
Hello, 

Sorry to revive an old thread, but I recently came across this project when reading into the recently releases clicks keyboard. Is something like a flat keyboard still possible? I've never soldered or built my own mechanical keyboard, but I'm following the guide and going to try to at least get the keyboard built and was really hoping to get something more flat, unless the current default way works better. 

Thanks!
